src/include/duckdb/parser/constraint.hpp

```diff
@@ -28,6 +29,24 @@ enum class ConstraintType : uint8_t {
 	FOREIGN_KEY = 4 // FOREIGN KEY constraint
 };
 
+enum class ForeignKeyType : uint8_t {
+	FK_TYPE_PRIMARY_KEY_TABLE = 0,   // main table
+	FK_TYPE_FOREIGN_KEY_TABLE = 1,   // referencing table
+	FK_TYPE_SELF_REFERENCE_TABLE = 2 // self refrencing table
```

Comment:
Does this need to be a different property? Would it not be sufficient to have two constraints in the same table (both `FK_TYPE_PRIMARY_KEY_TABLE` and `FK_TYPE_FOREIGN_KEY_TABLE`)?